flow/designs/asap7/mock-array-big/README.md

```diff
@@ -33,3 +33,63 @@ Now build the design as usual:
 ```
 make
 ```
+
+Element/constraint.sdc
+----------------------
+
+A single macro is made for all the elements in the array, though strictly
+speaking each macro is unique in that the timing constraints of each element
+in the array is different.
+
+To illustrate, consider the `assign io_lsbOuts_0 = io_lsbIns_1;` statement in Element.v.
+Here a signal is routed from left to right, through the Element without the signal
+being registered.
+
+Clearly, the maximum input delay for `io_lsbIns_1` is
+smaller for the leftmost than for the rightmost Element.
+
+Taking a step back, it is also worth considering what level of detail is
+appropriate for the Element's constraint.sdc file.
+
+In the beginning of a project, during exploration, the .sdc file does not
+need to be particularly detailed. In fact, details can be counterproductive.
```

Comment:
Missing a "why" here so I threw this in:

```suggestion
need to be particularly detailed. In fact, details can be counterproductive, as they tend to be inaccurate and cause the tools to spend time solving potential non-issues.
```

Reply:
Each step within the implementation flow will be related to different timing "magnitude/concerns", I think we need to segregate (separate by concerns) between different timing concerns within each step of the flow and/vs the tools accuracy.